src/pebble.py

```diff
@@ -73,6 +73,22 @@ def replan_mjolnir(self, container: ops.model.Container) -> None:
         container.add_layer("synapse-mjolnir", self._mjolnir_pebble_layer, combine=True)
         container.replan()
 
+    def replan_stats_exporter(self, container: ops.model.Container) -> None:
+        """Replan Synapse StatsExporter service.
+
+        Args:
+            container: Charm container.
+        """
+        access_token = self._charm_state.synapse_config.admin_access_token
+        if not access_token:
+            logger.warning("Admin Access Token is none, skipping Stats Exporter start.")
+        layer = self._stats_exporter_pebble_layer
+        layer["services"][synapse.STATS_EXPORTER_SERVICE_NAME]["environment"] = {
```

Comment:
It would make more sens to me to have that in the function that creates the stats_exporter layer. But then it couldn't be a property anymore. Why are we defining those create-layer-functions as properties ?

Reply:
IMHO The main reason for being properties is to dynamically create the environment, for example, every time the attribute is called. I think there is no significant change in performance but makes the code easier to read.

Reply:
I'm not sure that I like the property decorator in those cases but it's a stylistic choice (I'm not sure that I like decorators in general).